moving UI support to build in consul UI

sstarcher opened the pull request:
I removed the UI download and added the flag for -ui

I noticed -ui and -server are specified on the command line and the config.  These are duplicate settings would it make sense to remove them from the config and just specify them on the command line or vice versa?

sstarcher commented:
In reference to #120

progrium commented:
Ah, you're right. I think server should be specified on the command line to make it a little more obvious what's going on (server mode is the key differentiating feature of this image) when you read the Dockerfile. It doesn't need to be set in the config. Whereas UI makes more sense to be enabled just in the config like everything else.

sstarcher commented:
@progrium I updated it with that in mind.  So -server on the command line and ui specified in the config.  I also removed server from the config as it's on the command line.
